src/quacc/recipes/espresso/dos.py

```diff
@@ -0,0 +1,182 @@
+"""
+This module, 'dos.py', contains recipes for performing phonon calculations using the
+dos.x binary from Quantum ESPRESSO via the quacc library.
+
+The recipes provided in this module are jobs and flows that can be used to perform
+dos calculations.
+"""
+from __future__ import annotations
+
+from pathlib import Path
+from typing import TYPE_CHECKING
+
+from ase.io.espresso import Namelist
+
+from quacc import flow, job
+from quacc.calculators.espresso.espresso import EspressoTemplate
+from quacc.recipes.espresso._base import base_fn
+from quacc.recipes.espresso.core import non_scf_job, static_job
+from quacc.utils.dicts import recursive_dict_merge
+from quacc.wflow_tools.customizers import customize_funcs
+
+if TYPE_CHECKING:
+    from typing import Any, Callable
+
+    from ase.atoms import Atoms
+
+    from quacc.schemas._aliases.ase import RunSchema
+
+
+@job
+def dos_job(
+    prev_dir: str | Path,
+    parallel_info: dict[str] | None = None,
+    test_run: bool = False,
+    **calc_kwargs,
+) -> RunSchema:
+    """
+    Function to carry out a basic dos.x calculation (density of states).
+    It is mainly used to extract the charge density and wavefunction from a previous pw.x calculation.
+    It generates the total density of states. Fore more details please see
+    https://www.quantum-espresso.org/Doc/INPUT_DOS.html
+
+    Parameters
+    ----------
+    prev_dir
+        Outdir of the previously ran pw.x calculation. This is used to copy
+        the entire tree structure of that directory to the working directory
+        of this calculation.
+    parallel_info
+        Dictionary containing information about the parallelization of the
+        calculation. See the ASE documentation for more information.
+    **calc_kwargs
+        calc_kwargs dictionary possibly containing the following keys:
+
+        - input_data: dict
+        - additional_fields: list[str] | str
+
+        See the docstring of ase.io.espresso.write_fortran_namelist for more information.
+
+    Returns
+    -------
+    RunSchema
+        Dictionary of results from [quacc.schemas.ase.summarize_run][].
+        See the type-hint for the data structure.
+    """
+
+    calc_defaults = {"input_data": {"dos": {"fildos": "total_dos.dos"}}}
```

Comment:
By default QE reads all the data from the non scf xml file, so it sets defaults value taken from your previous calculations.

Reply:
fildos should be set to nothing because the default QE behaviour is to set it to "{prefix}.dos", it will be easier in the future to just copy everything that has {prefix} in it